Simplenote/src/main/java/com/automattic/simplenote/authentication/SignupFragment.java

```diff
@@ -0,0 +1,264 @@
+        initSignupButton(view);
+    }
+
+    @SuppressWarnings("ConstantConditions")
+    private void initSignupButton(View view) {
+        EditText emailEditText = ((TextInputLayout) view.findViewById(R.id.input_email)).getEditText();
+        final Button signupButton = view.findViewById(R.id.button);
+        setButtonState(signupButton, emailEditText.getText());
+        listenToEmailChanges(emailEditText, signupButton);
+        listenToSignupClick(signupButton, emailEditText);
+    }
+
+    private void listenToEmailChanges(EditText emailEditText, final Button signupButton) {
+        emailEditText.addTextChangedListener(new TextWatcher() {
+            @Override
+            public void afterTextChanged(Editable s) {
+                setButtonState(signupButton, s);
+            }
+
+            @Override
+            public void beforeTextChanged(CharSequence s, int start, int count, int after) {
+            }
+
+            @Override
+            public void onTextChanged(CharSequence s, int start, int before, int count) {
+            }
+        });
+    }
+
+    private void listenToSignupClick(Button signupButton, final EditText emailEditText) {
+        signupButton.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                if (NetworkUtils.isNetworkAvailable(requireContext())) {
+                    showProgressDialog();
+                    signupUser(emailEditText.getText().toString());
+                } else {
+                    showDialogError(getString(R.string.simperium_dialog_message_network));
+                }
+            }
+        });
+    }
+
+    private void showProgressDialog() {
+        progressDialogFragment =
+            ProgressDialogFragment.newInstance(getString(R.string.simperium_dialog_progress_signing_up));
+        progressDialogFragment.setStyle(DialogFragment.STYLE_NO_TITLE, R.style.Simperium);
+        progressDialogFragment.show(requireFragmentManager(), ProgressDialogFragment.TAG);
+    }
+
+    private void hideDialogProgress() {
+        if (progressDialogFragment != null && !progressDialogFragment.isHidden()) {
+            progressDialogFragment.dismiss();
+            progressDialogFragment = null;
+        }
+    }
+
+    private void signupUser(String email) {
+        new OkHttpClient()
+            .newBuilder()
+            .readTimeout(TIMEOUT_SECS, TimeUnit.SECONDS)
+            .build()
+            .newCall(buildCall(email))
+            .enqueue(buildCallback(email));
+    }
+
+    private Request buildCall(String email) {
+        return new Request.Builder()
+            .url(buildUrl())
+            .post(buildJsonBody(email))
+            .header(ACCEPT_LANGUAGE, getLanguage())
+            .build();
+    }
+
+    private RequestBody buildJsonBody(String email) {
+        JSONObject json = new JSONObject();
+        try {
+            json.put(JSON_USERNAME, email);
```

Comment:
is `JSON_USERNAME` necessary or does it get in the way of comprehension? I personally find it a small nuisance to read a name like that, jump to the top of the file, and then realize that the value matches the identifier. In these situations I like to ask myself, what if the JSON field were renamed? would we end up coming in here and renaming the constant as well? If the constant is 1:1-bound to the value, why not use the value and de-abstract, making it clearer when reading what JSON is being produced?

```java
json.put("username", email);
```

Reply:
This is done here: 75965c66f31de062b473afb61866c9e2b152d099